dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyFrameworkImpl.java

```diff
@@ -304,37 +300,53 @@ public ChannelFuture start(BootstrapExtended bootstrap, String inetHost, int ine
         return UDPUtils.start(this, bootstrap, inetHost, inetPort, bindListener);
     }
 
+    @Override
+    public ChannelFuture startOutbound(BootstrapExtended bootstrap, String inetHost, int inetPort,
+            ChannelFutureListener bindListener) throws NettyException {
+        if (bootstrap.getConfiguration() instanceof TCPConfigurationImpl) {
+            return TCPUtils.startOutbound(this, bootstrap, inetHost, inetPort, bindListener);
+        } else {
+            return UDPUtils.startOutbound(this, bootstrap, inetHost, inetPort, bindListener);
+        }
+    }
+
     @Override
     public ChannelFuture stop(Channel channel) {
-        if (isActive && channel.isOpen()) {
-            ChannelFuture closeFuture = channel.close();
-            activeChannels.remove(closeFuture.channel());
-            closeFuture.addListener(future -> {
-                if (!future.isSuccess()) {
-                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                        String channelName = channel.attr(ConfigConstants.NameKey).get();
-                        Tr.debug(tc,
-                                "channel " + channelName + " failed to stop due to: " + future.cause().getMessage());
+        if (activeChannels.contains(channel)) {
+            activeChannels.remove(channel);
+            if (isActive && channel.isOpen()) {
+                ChannelFuture closeFuture = channel.close();
+                closeFuture.addListener(future -> {
+                    if (!future.isSuccess()) {
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            String channelName = channel.attr(ConfigConstants.NameKey).get();
+                            Tr.debug(tc,
+                                    "channel " + channelName + " failed to stop due to: " + future.cause().getMessage());
+                        }
+                    } else {
+                        logChannelStopped(channel);
                     }
-                } else {
-                    logChannelStopped(channel);
-                }
-            });
-            return closeFuture;
+                });
+                return closeFuture;
+            } else {
+                logChannelStopped(channel);
+                return null;
+            }
         } else {
             return null;
         }
     }
 
     @Override
     public void stop(Channel channel, long timeout) {
-        if (this.isActive && channel.isOpen()) {
+        if (activeChannels.contains(channel)) {
```

Comment:
There is a race here perhaps if activeChannels membership is used as the gate - is it atomic etc?

Reply:
activeChannels is a ConcurrentHashMap which is thread safe so think we should be covered

Reply:
OK